web/src/features/fireWeather/components/graphs/PercipGraph.tsx

```diff
@@ -0,0 +1,153 @@
+import { NoonForecastValue } from 'api/forecastAPI'
+import { ToggleValues } from 'features/fireWeather/components/graphs/useGraphToggles'
+import {
+  defaultLayoutConfig,
+  findMaxNumber,
+  populateGraphDataForPrecip,
+  populateNowLineData
+} from 'features/fireWeather/components/graphs/plotlyHelper'
+
+const observedPrecipColor = '#fb0058'
+const forecastPrecipColor = '#a50b41'
+const hrdpsPrecipColor = '#a017c2'
+const rdpsPrecipColor = '#ea6d0e'
+const gdpsPrecipColor = '#f56c9c'
+
+interface Props {
+  currDate: Date
+  sliderRange: [string, string]
+  toggleValues: ToggleValues
+  observedValues: ObservedValue[]
+  forecastValues: NoonForecastValue[]
+  hrdpsModelValues: ModelValue[]
+  rdpsModelValues: ModelValue[]
+  gdpsModelValues: ModelValue[]
+}
+
+const PrecipGraph = (props: Props) => {
+  const {
+    currDate,
+    sliderRange,
+    toggleValues,
+    observedValues,
+    forecastValues,
+    gdpsModelValues,
+    rdpsModelValues,
+    hrdpsModelValues
+  } = props
+
+  const observation = populateGraphDataForPrecip(
+    observedValues,
+    'Observation',
+    observedPrecipColor,
+    toggleValues.showObservations
+  )
+  const forecast = populateGraphDataForPrecip(
+    forecastValues,
+    'Forecast',
+    forecastPrecipColor,
+    toggleValues.showForecasts
+  )
+  const hrdps = populateGraphDataForPrecip(
+    hrdpsModelValues,
+    'HRDPS',
+    hrdpsPrecipColor,
+    toggleValues.showHrdps
+  )
+  const gdps = populateGraphDataForPrecip(
+    gdpsModelValues,
+    'GDPS',
+    gdpsPrecipColor,
+    toggleValues.showGdps
+  )
+  const rdps = populateGraphDataForPrecip(
+    rdpsModelValues,
+    'RDPS',
+    rdpsPrecipColor,
+    toggleValues.showRdps
+  )
+
+  const maxY = findMaxNumber([
+    observation.maxAccumPrecip,
+    forecast.maxAccumPrecip,
+    hrdps.maxAccumPrecip,
+    gdps.maxAccumPrecip,
+    rdps.maxAccumPrecip
+  ])
+  const y2Range = [0, maxY]
+  const nowLine = populateNowLineData(currDate, y2Range[0], y2Range[1], 'y2')
+
+  return (
+    <div data-testid="precipitation-graph">
+      <Plot
+        style={{ width: '100%', height: '100%' }}
+        config={{ responsive: true }}
+        data={[
+          nowLine,
+          gdps.accumPrecipsline,
+          gdps.dailyPrecipsBar,
+          rdps.accumPrecipsline,
+          rdps.dailyPrecipsBar,
+          hrdps.accumPrecipsline,
+          hrdps.dailyPrecipsBar,
+          forecast.accumPrecipsline,
+          forecast.dailyPrecipsBar,
+          observation.accumPrecipsline,
+          observation.dailyPrecipsBar
+        ]}
+        layout={{
+          ...defaultLayoutConfig,
+          title: {
+            text: 'Daily Precipitation (with accumulated)',
+            yanchor: 'middle'
+          },
+          barmode: 'group',
+          bargap: 0.75,
+          bargroupgap: 0.3,
+          xaxis: {
+            range: sliderRange,
+            rangeslider: {
+              visible: true,
+              bgcolor: '#dbdbdb',
```

Comment:
This value is repeated for precipitation, temprh and wind - maybe it should be a single constant somewhere? Probably deserves at least being a constant at the top of this file?

Reply:
I created a shared object for the range slider config